x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/WaitForRolloverReadyStep.java

```diff
@@ -53,6 +55,12 @@ public void evaluateCondition(IndexMetaData indexMetaData, Listener listener) {
             return;
         }
 
+        if (indexMetaData.getSettings().getAsBoolean(LIFECYCLE_ROLLOVER_SKIP_ROLLED, true) &&
+            indexMetaData.getRolloverInfos().get(rolloverAlias) != null) {
+            logger.trace("Index {} was already rolled, skipping", indexMetaData.getIndex().getName());
+            return;
```

Comment:
This should actually be:

```java
listener.onResponse(true, new WaitForRolloverReadyStep.EmptyInfo());
```

To simulate that the condition has been met, otherwise we'll idle in this step forever